iot-hub/Samples/device/ImportExportDevicesSample/ProcessImpExpCommands.cs

```diff
@@ -0,0 +1,438 @@
+        /// <param name="num_to_add">Number of devices to add.</param>
+        public async static Task AddDevicesToHub(int num_to_add)
+        { 
+            string URI_to_container = PrepareForImportExport();
+            // Add num_to_add devices to the IoT Hub.
+            // This won't change any devices already registered with the hub, it just adds more.
+            // The default is 10. 
+            if (num_to_add < 0)
+            {
+                num_to_add = 10;
+            }        
+            await IOTHubDevices.GenerateAndAddDevices(URI_to_container, num_to_add);
+        }
+
+        /// <summary>
+        /// Read the list of registered devices from the IoT hub and export it to blob storage.
+        /// </summary>
+        public async static Task ExportDevicesToBlobStorage()
+        {
+            string URI_to_container = PrepareForImportExport();
+            await IOTHubDevices.ExportDevices(URI_to_container);
+        }
+
+        /// <summary>
+        /// Read the device list that was exported to blob storage,
+        ///   then print it to the console.
+        /// The point of this is just so you can see the data.
+        /// </summary>
+        /// <returns></returns>
+        public async static Task ReadAndDisplayExportedDeviceList()
+        {
+            string URI_to_container = PrepareForImportExport();
+            await IOTHubDevices.ReadExportedDeviceList();
+        }
+
+        /// <summary>
+        /// Read a list of devices from IoT Hub, then delete them. 
+        /// This version of this deletes all devices registered to the hub. 
+        /// </summary>
+        /// <returns></returns>
+        public async static Task DeleteAllDevicesFromHub()
+        {
+            string URI_to_container = PrepareForImportExport();
+            await IOTHubDevices.DeleteDevices(URI_to_container);
+        }
+
+        //generate num_to_add devices and add them to the hub 
+        // to do this, generate each identity 
+        // * include authentication keys
+        // * write the device info to a block blob
+        // * import the devices into the identity registry by calling the import job
+        private static async Task GenerateAndAddDevices(string URI_to_container, int num_to_add)
+        {
+            //generate reference for list of new devices you're going to add, will write list to this blob 
+            CloudBlockBlob generatedListBlob = cloudBlobContainer.GetBlockBlobReference(devices_to_add);
+
+            // define serializedDevices as a generic list<string>
+            List<string> serializedDevices = new List<string>();
+            
+            for (var i = 1; i <= num_to_add; i++)
+            {
+                // Create device name with this format: Hub_00000000 + a new guid.
+                // This should be large enough to display the largest number (1 million).
+                string deviceName = "Hub_" + i.ToString("D8") + "-" + Guid.NewGuid().ToString();
```

Comment:
I think this would be better to use string.format or string interpolation `string deviceName = $"Hub_{i.ToString("D*")}-{Guid.NewGuid().ToString();`

Reply:
okay, thanks. I put It in the expanded format when I was debugging; I've fixed it now.